src/Microsoft.TemplateEngine.Authoring.TemplateVerifier/VerificationEngine.cs

```diff
@@ -0,0 +1,335 @@
+            Verifier.DerivePathInfo(
+                (sourceFile, projectDirectory, type, method) => new(
+                    directory: options.ExpectationsDirectory ?? "VerifyExpectations",
+                    typeName: options.TemplateName,
+                    methodName: EncodeArgsAsPath(options.TemplateSpecificArgs)));
+
+            if ((options.UniqueFor ?? UniqueForOption.None) != UniqueForOption.None)
+            {
+                foreach (UniqueForOption value in Enum.GetValues(typeof(UniqueForOption)))
+                {
+                    if ((options.UniqueFor & value) == value)
+                    {
+                        switch (value)
+                        {
+                            case UniqueForOption.None:
+                                break;
+                            case UniqueForOption.Architecture:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForArchitecture();
+                                break;
+                            case UniqueForOption.OsPlatform:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForOSPlatform();
+                                break;
+                            case UniqueForOption.Runtime:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForRuntime();
+                                break;
+                            case UniqueForOption.RuntimeAndVersion:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForRuntimeAndVersion();
+                                break;
+                            case UniqueForOption.TargetFramework:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForTargetFramework();
+                                break;
+                            case UniqueForOption.TargetFrameworkAndVersion:
+                                defaultVerifyTask = defaultVerifyTask.UniqueForTargetFrameworkAndVersion();
+                                break;
+                            default:
+                                throw new ArgumentOutOfRangeException();
+                        }
+                    }
+                }
+            }
+
+            if (options.DisableDiffTool ?? false)
+            {
+                defaultVerifyTask = defaultVerifyTask.DisableDiff();
+            }
+
+            return defaultVerifyTask;
+        }
+
+        private async Task VerifyResult(TemplateVerifierOptions args, CommandResult commandResult)
+        {
+            // Customize diff output of verifier
+            VerifyDiffPlex.Initialize(OutputType.Compact);
+
+            UsesVerifyAttribute a = new UsesVerifyAttribute();
+            // https://github.com/VerifyTests/Verify/blob/d8cbe38f527d6788ecadd6205c82803bec3cdfa6/src/Verify.Xunit/Verifier.cs#L10
+            //  need to simulate execution from tests
+            var v = DummyMethod;
+            MethodInfo mi = v.Method;
+            a.Before(mi);
```

Comment:
@SimonCropp is there any way we can avoid this workaround?

Reply:
@vlada-shubina i can give u static `SetMethodInfo` api on UsesVerifyAttribute. it would need to be called on the same thread before a verify occurs. or i can give u an optional `MethodInfo` parameter on all the `Verify*` methods.

Reply:
@SimonCropp I'd say it's still very similar situation - a dummy `MethodInfo` needs to be passed even though it doesn't end up being used (we need to name directories/files by scenarios run, not by the code that was invoking them - so the type and method name are [overriden by `PathInfo`](https://github.com/VerifyTests/Verify/blob/main/src/Verify/Naming/FileNameBuilder.cs#L5-L7) and parameters are not used as method is intentionally parameterless).
In ideal case the [`InnerVerifier`](https://github.com/VerifyTests/Verify/blob/main/src/Verify.Xunit/Verifier.cs#L10) would pass `Lazy` that would throw only on access (plus there would need to be a way to indicate that parameters should not be used).
Though our scenario is pretty specific (and easily workaroundable) - so I understand this might not meet the bar.